### ISSUE TITLE
Reference-style links to .md files aren't resolved to the generated url

### ISSUE THREAD

**Issue by Firehed** [CouscousPHP/Couscous#155]
Using a [reference-style link](https://daringfireball.net/projects/markdown/syntax#link) fails to automatically convert a markdown file url to its html counterpart, which is inconsistent with the [documentation](http://couscous.io/docs/faq.html) indicating that links should work between pages.

That is to say:

```markdown
Using [a link](foo.md) to [a page][bar] or [baz][]

[bar]: bar.md
[baz]: baz.md
```
renders
```html
Using <a href="foo.html">a link</a> to
<a href="bar.md">a page</a> or <a
href="baz.md">baz</a>
```
instead of 
```html
Using <a href="foo.html">a link</a> to
<a href="bar.html">a page</a> or <a
href="baz.html">baz</a>
```

In practice it's not really a huge issue, but it's not quite what you would expect to happen.

**Comment by mnapoli** [comment]
Right, I'm guessing that [this class](https://github.com/CouscousPHP/Couscous/blob/master/src/Module/Markdown/Step/RewriteMarkdownLinks.php) only rewrites classic links and not reference-style links. I'm guessing #134 (i.e. replace Parsedown with CommonMark) would help since we could get rid of the regex and directly edit the AST that was generated from the markdown source. But for that we need to have a CommonMark extension for Markdown Extra, because currently that's a feature offered by Couscous.

**Comment by mnapoli** [comment]
On the "CommonkMark" topic I've opened #156.
